include/exadg/fluid_structure_interaction/precice/driver_fluid.h

```diff
@@ -218,7 +216,32 @@ class DriverFluid : public Driver<dim, Number>
 
       // post-solve
       if(is_new_time_window)
+      {
+        // computes new time-step size
         fluid->time_integrator->advance_one_timestep_post_solve();
+        // next, we synchronize the time-step sizes. Have a look in the base class for more
+        // explanations. Subcycling would in theory be compatible in explicit coupling schemes here.
+        // In implicit coupling schemes, we need matching time-window sizes as the time-step size
+        // push back happens in the 'advance_one_timestep_post_solve()' and we cannot change two
+        // subsequent time-step sizes without a push back operation, otherwise we falsify the time
+        // integrator. However, the treatment of the boundary conditions is not handled correctly
+        // when using subcycling, as we would apply the 'whole displacement' within the first
+        // time-step (being potentially only a fractio of the whole time-windo). In order to cope
```

Comment:
why "potentially"? Doesn't "subcycling" mean that one time-window consists of several time steps? 

There are some typos, but I would remove what is written in the brackets since this is redundant with the "nomenclature" comment.

Reply:
> why "potentially"? Doesn't "subcycling" mean that one time-window consists of several time steps?

Sure. With time-step adaptivity you don't know in advance weather the time-step size is equal to the time-window size, though. I deleted the comment in brackets now.